internal/plugin/connectors/http/generic/config_test.go

```diff
@@ -59,6 +64,32 @@ func Test_newConfig(t *testing.T) {
 		// Assert against value calculated with independent base64 encoder.
 		assert.Equal(t, "Basic Sm9uYWg6c2VjcmV0", headers["Authorization"])
 	})
+
+	t.Run("creates expected query params", func(t *testing.T) {
+		cfg, err := sampleConfig()
+		assert.NoError(t, err)
+		if err != nil {
+			return
+		}
+
+		params, err := render(
+			cfg.QueryParams,
+			map[string][]byte{
+				"key": []byte("someKey"),
+				"name": []byte("Foo Bar"),
+			})
+
+		assert.NoError(t, err)
+		if err != nil {
+			return
+		}
+
+		// A simple test case
+		assert.Equal(t, "Foo Bar", params["NameWithSpaces"])
+
+		// Assert against value calculated with independent base64 encoder.
+		assert.Equal(t, "c29tZUtleQ==", params["Key"])
+	})
```

Comment:
I'd like to see a couple of more tests here:
- One with special/utf chars so that we know encode is working properly
- One with existing url params so that we can be sure that appending works